plugins/CoreHome/Controller.php

```diff
@@ -111,7 +111,7 @@ public function showInContext()
         $controllerName = Common::getRequestVar('moduleToLoad');
         $actionName     = Common::getRequestVar('actionToLoad', 'index');
 
-        if($controllerName == 'API') {
+        if($controllerName == 'API' || ($controllerName == 'Proxy' && $actionName == 'callApi')) {
             throw new Exception("Showing API requests in context is not supported for security reasons. Please change query parameter 'moduleToLoad'.");
```

Comment:
I understand your point, but atm we aren't really using custom exceptions in nearly all cases. I think we need to make a general decision in core team if we would like to change that in the future.
But even if so I wouldn't start doing that before Piwik 3.0

cc @mattab @tsteur